README.md

```diff
@@ -291,6 +291,105 @@ Person.getOverTwenties() { result, error in
+    // Use person and id
+}
+```
+
+### Manual id assignment
+
+You can add an id field to your model and manage the assignment of id’s yourself by either specifying a property name for your id field using the `idColumnName` property or by defining a field named `id`, which is the default name for the aforementioned property. For example:
+
+```swift
+struct Person: Model {
+    var myIDField: Int
+    var firstname: String
+    var surname: String
+    var age: Int
+
+    static var idColumnName = "myIDField"
+    static var idColumnType = Int.self
+}
+```
+
+When using a `Model` defined in this way all the id management is the responsibility of the user. An example of saving an instance of this Person is below:
+
+```swift
+let person = Person(myIDField: 1, firstname: "example", surname: "person", age: 21)
+person.save() { person, error in
+    guard let person = person else{
+        // Handle error
+        return
+    }
+    // Use newly saved person
+}
+```
+
+### Using `optional` id fields
+
+If you would like an id field that allows you to specify specific values whilst also being automatic when an id is not explicitly set you can use an optional Int for your id field and re-define the `idKeypath` property to point at the field. `IDKeyPath` is a typealias for `WritableKeyPath<Self, Int?>?`, your re-definition must be explicity set this type. For example:
+
+```swift
+struct Person: Model {
+    var id: Int?
+    var firstname: String
+    var surname: String
+    var age: Int
+
+    static var idKeyPath: IDKeyPath = \Person.id
+}
+```
+
+In the above example the `Model` is defined with an ID field matching the default `idColumnName` property, should you wish to use an alternative name you need to re-define `idColumnName` accordingly. The `optional` id field is limited to the `Int` type.
+
+When saving an instance of this `Person` you can either set a specific value for `id` or set `id` to `nil`, for example:
+
+```swift
+let person = Person(id: nil, firstname: “Banana”, surname: “Man”, age: 21)
+let otherPerson = Person(id: 5, firstname: “Super”, surname: “Ted”, age: 26)
+
+person.save() { savedPerson, error in
+        guard let newPerson = savedPerson else {
+            // Handle error
+        }
+        print(newPerson.id) // Prints the next value in the databases identifier sequence, for the first save this will be 1
+}
+
+otherPerson.save() { savedPerson, error in
+        guard let newOtherPerson = savedPerson else {
+            // Handle error
+        }
+        print(newOtherPerson.id) // Prints 5
+}
+```
+
+**NOTE** - When using manual or option id fields you need to ensure that the application is written to handle violation of unique identifier constraints which will occur if you attempt to save a model with an id that already exists in the database. If you are using `SwiftKueryPostgreSQL` this error can occur when savinf with a `nil` value for the id property due to the way PostgreSQL implements `autoincrement` behaviour.
```

Comment:
typo: saving
Also, if we're going to mention Postgres' behaviour specifically, we might want to elude to why (ie. that it does not skip over existing ids while incrementing).